src/wp-admin/includes/update-core.php

```diff
@@ -1178,30 +1178,25 @@ function update_core( $from, $to ) {
 		);
 	}
 
-	// Add a warning when the JSON PHP extension is missing.
-	if ( ! extension_loaded( 'json' ) ) {
-		return new WP_Error(
-			'php_not_compatible_json',
-			sprintf(
-				/* translators: 1: WordPress version number, 2: The PHP extension name needed. */
-				__( 'The update cannot be installed because WordPress %1$s requires the %2$s PHP extension.' ),
-				$wp_version,
-				'JSON'
-			)
-		);
+	$missing_extensions = new WP_Error();
+
+	foreach ( $required_php_extensions as $extension ) {
```

Comment:
Is it worth doing an `isset` here in case something weird happens, such as a failed upgrade, to make sure required php extensions is defined.

It's probably worht checking when the failed upgrade is caught.